external_search.py

```diff
@@ -633,14 +666,15 @@ def bulk_update(self, works, retry_on_batch_failure=True):
         
         # We weren't able to create search documents for these works, maybe
         # because they don't have presentation editions yet.
-        missing_works = [work for work in works if work.id not in doc_ids]
-            
-        error_ids = [
-            error.get('data', {}).get("_id", None) or
-            error.get('index', {}).get('_id', None)
-            for error in errors
-        ]
+        def get_error_id(error):
+            return error.get('data', {}).get('_id', None) or error.get('index', {}).get('_id', None)   
+        error_ids = [get_error_id(error) for error in errors]
 
+        missing_works = [
+            work for work in works 
+            if work.id not in doc_ids and work.id not in error_ids
+        ]
+            
```

Comment:
Previously we counted a work both as 'error' and as 'missing' if it caused an error.